docs/releases/changelog-dev.md

```diff
@@ -4,6 +4,9 @@ This document contains the changes of the current release.
 
 ### New features since last release
 
+- Changed gate settings serialization so that fields with None values are not in the resulting dictionary
+  [#562](https://github.com/qilimanjaro-tech/qililab/pull/562)
+
```

Comment:
There should be a test testing this!!

Reply:
there is! the runcard in data.galadriel has
https://github.com/qilimanjaro-tech/qililab/blob/18b269e61dabfabaf6f32548a0aabf3c9bcab8ab/tests/data.py#L142-L163
so the object created in `platform.gate_settings.gates` will have one gate containing options, the other containing `None`. If the serialization works as intended then the test at
https://github.com/qilimanjaro-tech/qililab/blob/18b269e61dabfabaf6f32548a0aabf3c9bcab8ab/tests/settings/test_runcard.py#L40C87-L40C87
will pass, otherwise it won't because it will have the extra field `{options: None}` for gate `CZ(0, 2)`

Reply:
the problem is that I originally didn't add these `options` field in the `tests` runcard

Reply:
Nice! I forgot about this test!!